data-collection/src/main/java/refactoringml/ProcessMetricsCollector.java

```diff
@@ -51,73 +47,44 @@ public ProcessMetricsCollector(Project project, Database db, Repository reposito
 		this.branch = branch;
 		this.fileStoragePath = FilePathUtils.lastSlashDir(fileStoragePath);
 		this.lastCommitToProcess = lastCommitToProcess;
-		todo = new HashMap<>();
 		pmDatabase = new PMDatabase(commitThreshold);
 	}
 
-	public void addToList (RevCommit commitData, Yes yes) {
-		String id = commitData.getName();
-		if(!todo.containsKey(id))
-			todo.put(id, new HashSet<>());
-
-		todo.get(id).add(yes.getId());
-	}
-
-	public void collect() throws IOException {
-
-		RevWalk walk = JGitUtils.getReverseWalk(repository, branch);
-
-		RevCommit commit = walk.next();
-
-		boolean lastFound = false;
-		while(commit!=null && !lastFound) {
-
-			// did we find the commit to stop?
-			// if so, process it, and then stop
-			if(commit.getName().equals(lastCommitToProcess))
-				lastFound = true;
-
-			if(commit.getParentCount() <= 1) {
-
-				log.debug("Commit ID " + commit.getName());
-				RevCommit commitParent = commit.getParentCount() == 0 ? null : commit.getParent(0);
+	public void collectMetrics(RevCommit commit, Set<Long> allYeses, boolean isRefactoring) throws IOException {
+		if(commit.getParentCount() > 1) return;
```

Comment:
Just a note: maybe we should skip merge commits in the "main walker" in the App file. What do you think?

Reply:
And if we do that there, then there's no need to repeat it here